docs/bare_metal/bare_metal.md

```diff
@@ -0,0 +1,108 @@
+
+The bare metal profile implements a subset of Mbed OS's RTOS APIs that are useful in non-threaded applications, such as semaphores (calling the release API from interrupts) and tickers (to set up a recurring interrupt). It does not include an RTX and is not a full RTOS, and is therefore suitable for applications that do not require complex thread management. Instead of the RTOS's scheduler, all activities are polled or interrupt-driven. This simplifies application code and allows using APIs that are not thread safe. Just as important, you can use the code-optimized versions of the C standard libraries, `microlib` and `newlib-nano`, which are much smaller than the thread safe equivalents the full profile requires.
+
+## Features
+
+For a breakdown of supported APIs, please see [the full API list](../apis/index.html).
+
+The Mbed OS build tools - Mbed CLI, Mbed Online Compiler and Mbed Studio - all support working with the bare metal profile.
+
+<span class="notes">**Note:** Because bare metal uses some APIs that Mbed OS classifies as RTOS APIs, some class names that traditionally belong in RTOS programming are used in bare metal. For example, bare metal uses the class `ThisThread` despite not using threads.</span>
+
+<table>
+    <thead>
+        <tr>
+            <th colspan="2">Features</th>
+            <th>Support details</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan="6">Core</td>
+            <td>Drivers</td>
+            <td>Available; enabled by default</td>
+        </tr>
+        <tr>        
+            <td>Events</td>
+            <td>Available</td>
+        </tr>
+        <tr>        
+            <td>HAL</td>
+            <td>Available</td>
+        </tr>
+        <tr>        
+            <td>Platform</td>
+            <td>Available; enabled by default</td>
+        </tr>
+            <tr>        
+            <td>RTOS APIs</td>
+            <td>Semaphore, Mutex, EventFlags, ThisThread: Available; enabled by default.<br><br> All other APIs not supported.</td>
+        </tr>
+        <tr>        
+            <td>Storage</td>
+            <td>Available</td>
+        <tr>
+            <td rowspan="8">Connectivity</td>
+            <td>802.15.4_RF</td>
+            <td>Not Available</td>
+        </tr>
+        <tr>
+            <td>Wifi</td>
+            <td>Not Available</td>
+        </tr>
+        <tr>
+            <td>Cellular</td>
+            <td>Not Available</td>
+        </tr>
+        <tr>
+            <td>LWIP stack</td>
+            <td>Not Available</td>
+        </tr>
+        <tr>
+            <td>Nanostack</td>
+            <td>Not Available</td>
+        </tr>
+        <tr>
+            <td>Network Socket</td>
+            <td>Not Available</td>
+        </tr>
+        </tr>
+        <tr>
+            <td>BLE</td>
+            <td>Available<br>(Except on `TARGET_NORDIC_CORDIO`)</td>
```

Comment:
For completeness, currently two drivers are not supported: COMPONENT_BlueNRG_MS and TARGET_NORDIC_CORDIO. We should either list which ones are supported and which ones aren't or say that not all drivers are supported? FI, the cordio driver could be ported to bare metal next week.

Reply:
If there are only two we don't support, I think listing both is nice, as it removes the question "well if you say 'not all drivers' how do I know which ones".

Reply:
```suggestion
            <td>Available<br>Except `COMPONENT_BlueNRG_MS` and `TARGET_NORDIC_CORDIO`</td>
```

Reply:
Good (but the highlighter does not work)

Reply:
Okay, I'll figure the formatting out